server/bleep/src/indexes/file.rs

```diff
@@ -173,14 +189,41 @@ impl Indexable for File {
 
             let walker = FileWalker::index_directory(&repo.disk_path, branch);
             let count = walker.len();
+            analytics_event.add_payload("file_count", &count);
             walker.for_each(pipes, file_worker(count));
         };
 
         if pipes.is_cancelled() {
             bail!("cancelled");
         }
 
-        info!(?repo.disk_path, "repo file indexing finished, took {:?}", start.elapsed());
+        // aggregate stats
+        drop(worker_stats_tx);
+        let mut repo_stats = WorkerStats {
+            is_first_index,
+            was_index_reset: app.indexes.was_index_reset,
+            ..Default::default()
+        };
+        while let Ok(Some(stats)) =
+            tokio::time::timeout(Duration::from_millis(10), worker_stats_rx.recv()).await
+        {
+            repo_stats += stats;
+        }
+        analytics_event.add_payload("chunk_count", &repo_stats.chunks);
+        analytics_event.add_payload("bytes", &human_readable(repo_stats.size));
+        analytics_event.add_payload("index_job_kind", &repo_stats.index_job_kind());
+
+        let sync_time = start.elapsed();
+        info!(?repo.disk_path, "repo file indexing finished, took {:?}", sync_time);
+        analytics_event.add_payload("sync_time", &format!("{sync_time:?}"));
+
+        dbg!(&analytics_event);
+
+        // send an analytics event if this is the first time this repo is being indexed
+        if repo_stats.reindex_count > 0 {
+            let user = app.user().await;
+            app.with_analytics(|hub| hub.track_repo(analytics_event, &user));
+        }
```

Comment:
It may read nicer if we had an `indexes::analytics` (or something to that effect) module, which would contain this logic in a struct wrapped around the `rx` handle + all the other ancilliary structs. You could also measure index time between instantiation and commit as index time.